plugins/kafka/builtins/plugin.cpp

```diff
@@ -105,9 +129,33 @@ class load_plugin final
 
 class save_plugin final
   : public virtual operator_plugin2<saver_adapter<kafka_saver>> {
-  auto initialize(const record& config,
-                  const record& /* global_config */) -> caf::error override {
-    config_ = config;
+  auto initialize(const record& unused_plugin_config,
+                  const record& global_config) -> caf::error override {
+    if (not unused_plugin_config.empty()) {
+      return caf::make_error(ec::diagnostic, fmt::format("`{}.yaml` is unused; "
+                                                         "Use `kafka.yaml` "
+                                                         "instead",
+                                                         this->name()));
```

Comment:
Style question (for my understanding): isn't the common pattern to go through the `diagnostic::error(...)` factory and call `to_error()` at the end?

Reply:
Yes that would be better. TBH I simply copied that code from the `s3` plugin and replaced the name.